src/app/(guest)/profile/[token]/page.tsx

```diff
@@ -57,7 +57,9 @@ export default async function profile({ params }: ProfileProps) {
         Os teus <span className="text-primary">exames</span>
       </p>
 
-      <PreviousExamsTable user={user} />
+      <PreviousExamsTable user={user} token={params.token} />
```

Comment:
Fetch the data in parent component and just pass the data to the table. This enables the use of Promise.all()